### PR TITLE
Support LIMIT ALL.

### LINKED ISSUE #3158: Support Limit All
In some database like PostsgreSQL, MySQL "Limit All" has the same effect as eliminating the LIMIT clause from a query; it returns all the rows. Currently it gives error in Presto that the argument of Limit clause must be an integer.

### PR DESCRIPTION
**Pull request by yaoxin226**
LIMIT ALL is the same as omitting the LIMIT clause.
Fix #3158

### CODE REVIEW

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 1]
```diff
@@ -188,6 +188,7 @@ LIMIT Clause
 ------------
 
 The ``LIMIT`` clause restricts the number of rows in the result set.
+``LIMIT ALL`` is the same as omitting the LIMIT clause.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/sql/select.rst]: Add backticks around "LIMIT"

**Comment by electrum** [comment]
To be consistent, we should also support this for `SHOW PARTITIONS` as well.

**Comment by yaoxin226** [comment]
Should we support `SHOW PARTITIONS` in this PR, or create another PR?

**Comment by electrum** [comment]
Add it to this one, since it is part of the same change (extending syntax of `LIMIT`).

**Comment by yaoxin226** [comment]
Added `SHOW PARTITIONS` support.

**Comment by yaoxin226** [comment]
ping

**Comment by martint** [comment]
Merged, thanks!

**Comment by yaoxin226** [comment]
Thanks for the review.

**Pull request closed without merging**